api/v1alpha1/cassandrabackup_types.go

```diff
@@ -32,6 +32,9 @@ type CassandraBackupSpec struct {
 
 	// The name of the CassandraDatacenter to back up
 	CassandraDatacenter string `json:"cassandraDatacenter"`
+
+	//The type of the backup: "full" or "differential"
+	Type string `json:"backupType,omitempty"`
```

Comment:
How about we make this a bit more strongly-typed  with this:

```go
type BackupType string

const (
    FullBackup BackupType = "full"
    DifferentialBackup BackupType = "differential"
)
```
Then change the `Type` field to `BackupType` instead of string.

I also suggest adding a couple kubebuilder annotations so we end with:

```go
// +kubebuilder:validation:Enum=differential;full
// +kubebuilder:default="differential"
Type BackupType `json:"backupType,omitempty"`
```
What do you think?

Reply:
Sounds good... I'll make the update shortly :)